lib/vm/src/instance/mod.rs

```diff
@@ -397,7 +393,7 @@ impl Instance {
     }
 
     /// Invoke the WebAssembly start function of the instance, if one is present.
-    fn invoke_start_function(&self) -> Result<(), Trap> {
+    fn invoke_start_function(&self, trap_info: &dyn TrapInfo) -> Result<(), Trap> {
```

Comment:
I wonder whether it should be better to write this as such:

```suggestion
    fn invoke_start_function<T: TrapInfo>(&self, trap_info: &T) -> Result<(), Trap> {
```

Reply:
I tried that after reading your suggestion, had to revert.
In the `Engine` a trait the `finish_instantiation` function (which calls this function under the hood) would need to take a generic parameter which causes the code to not compile.